apps/web/src/studio/hooks/useStudioNavigate.ts

```diff
@@ -0,0 +1,25 @@
+import { useCallback } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { parseUrl } from '../../utils/routeUtils';
+
+export function useStudioNavigate() {
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
🤓 **nitpick (non-blocking):** Add `/** */` comment about what this does and why it's necessary‏